tt_metal/impl/device/device_pool.hpp

```diff
@@ -34,9 +38,11 @@ class DevicePool {
 
     Device *get_active_device(chip_id_t device_id) const;
     std::vector<Device *> get_all_active_devices() const;
-    bool close_device(chip_id_t device_id) const;
+    bool close_device(chip_id_t device_id);
     bool is_device_active(chip_id_t id) const;
-
+    void register_worker_thread_for_device(Device* device, uint64_t worker_thread_id);
+    void delete_worker_thread_for_device(Device* device);
```

Comment:
I assume these functions can be called from multiple different threads, is that true/false?
If thats false, that should be clearly stated in the contract of the device_pool/these methods.
If thats true, the implementation of these methods is currently not thread safe.